File: java-design-pattern/src/main/java/intro/to/design/patterns/using/javalang/factorymethod/creditcard/CreditCardFactory.java
package intro.to.design.patterns.using.javalang.factorymethod.creditcard;

import intro.to.design.patterns.using.javalang.factorymethod.framework.Factory;
import intro.to.design.patterns.using.javalang.factorymethod.framework.Product;

public class CreditCardFactory extends Factory {
    @Override
    protected Product createProduct(String owner) {
        return new CreditCard(owner);
    }

    @Override
    protected void registerProduct(Product product) {
        System.out.println("now registered " + product);
    }
}

File: typescript-design-pattern/src/into-to-design-patterns-using-java-lang/factory-method/credit-card/credit-card-factory.ts
import Factory from '../framework/factory'
import { Product } from '../framework/product'
import Output from '../../../utils/stdin-stdout/output'
import CreditCard from '.'

export default class CreditCardFactory extends Factory {
  protected createProduct(owner: string): Product {
    return new CreditCard(owner)
  }

  protected registerProduct(product: Product): void {
    Output.dir({ '->': 'now registered ', product })
  }
}
